extension/js/content_scripts/webmail/gmail_element_replacer.ts

```diff
@@ -550,4 +552,18 @@ export class GmailElementReplacer implements WebmailElementReplacer {
     }
   }
 
+  private addEndSessionBtnIfNeeded = async () => {
+    const keys = await Store.keysGet(this.acctEmail);
+    for(const key of keys) {
+      // Check if passpharse in the session
+      if (!(await Store.passphraseGet(this.acctEmail, key.longid, true)) &&  
```

Comment:
I recommend to put this logic into `Store.isPassPhraseCurrentlyInSession`

Reply:
Agree, thanks